common/persistence/persistence-tests/historyV2PersistenceTest.go

```diff
@@ -876,7 +876,8 @@ func (s *HistoryV2PersistenceSuite) append(branch []byte, events []*historypb.Hi
 	}
 	s.True(resp.Size > 0)
 
-	return err
+	// todo: the error is always nil here
+	return nil
```

Comment:
I'd just go ahead and remove this one and anything else glaringly wrong that you see, but there are a few other issues in this method. 

1) We don't call `s.T().Helper()` at the top which will make debugging assertions like line 877 harder. 
2) We're returning the unwrapped error in some cases here which means we won't have a stack trace at all. We should use pkg/errors and wrap these because these are tests where the performance hit will be negligible.
3) We're inconsistent about using test assertions vs. returning errors in this helper method. I think we should choose one approach and stick with it. Personally, I think we should always return errors wrapped with pkg/errors and run no assertions in our helper methods.